### ISSUE TITLE
400_MESSAGE_NOT_MODIFIED_ TRIGGER FAULTY RCLONE COPY

### ISSUE THREAD

**Issue by sevvian** [gautamajay52/TorrentLeech-Gdrive#106]
Hi

My Setup:
=======
I have installed torrentleech -gdrive in  raspberry pi through docker using Dockerfile provided . I have network connection speed of 60MBps Down and 15 MBps UP.

Issue:
====

Whenever i try to download bigger torrent files through torrentleech-gdrive, any filesize >=2GB i see some strange behaviour that during the torrent is being downloaded and not yet complete , Rclone gets triggered which results in Rclone copy fails because the file is being updated by aria download (file modified time keeps modifying because the aria2 still downloads the file in the background as it has not completed yet -)

My analysis:
=============
Even though its not because of the filesize (i tried multiple builds with different configs).   I see  that some torrents MAx out my connection speed and saturates the bandwidth limit and so the API calls/internet browsing experience delays / shows increased latency for API calls.

which causes Pyrogram 400 Message not modified in the logs (may be due to message edit timeouts )  (note:  I even increased the Edit_message_timeout = 120 or greater). But that did not help.

Please find the detail log below

[https://del.dog/torrentleech-gdrive-rclone-trigger-on-400-message-not-modified.txt](url)

Please fix the rclone trigger before the download completion through some sort of polling mechanism or API call to aria2 to make sure the download job completed , only then Rclone should start copy.

thanks,
MJ

**Comment by gautamajay52** [comment]
Can I know the value of your `EDIT_SLEEP_TIME_OUT` ?🤔

**Comment by gautamajay52** [comment]
> Hi
> 
> # My Setup:
> I have installed torrentleech -gdrive in raspberry pi through docker using Dockerfile provided . I have network connection speed of 60MBps Down and 15 MBps UP.
> 
> # Issue:
> Whenever i try to download bigger torrent files through torrentleech-gdrive, any filesize >=2GB i see some strange behaviour that during the torrent is being downloaded and not yet complete , Rclone gets triggered which results in Rclone copy fails because the file is being updated by aria download (file modified time keeps modifying because the aria2 still downloads the file in the background as it has not completed yet -)
> 
> # My analysis:
> Even though its not because of the filesize (i tried multiple builds with different configs). I see that some torrents MAx out my connection speed and saturates the bandwidth limit and so the API calls/internet browsing experience delays / shows increased latency for API calls.
> 
> which causes Pyrogram 400 Message not modified in the logs (may be due to message edit timeouts ) (note: I even increased the Edit_message_timeout = 120 or greater). But that did not help.
> 
> Please find the detail log below
> 
> [Log File ](https://del.dog/torrentleech-gdrive-rclone-trigger-on-400-message-not-modified.txt)
> 
> Please fix the rclone trigger before the download completion through some sort of polling mechanism or API call to aria2 to make sure the download job completed , only then Rclone should start copy.
> 
> thanks,
> MJ

And also as per logs...your config is not correct, I guess🤔

**Comment by sevvian** [comment]
I have the same config , it works for files less than 2gb , saying it works well  and uploads properly to my drive. I dont have issues with small filesizes of <2gb.. I have issues with bigger files and download takes much longer

**Comment by gautamajay52** [comment]
> I have the same config , it works for files less than 2gb , saying it works well and uploads properly to my drive. I dont have issues with small filesizes of <2gb.. I have issues with bigger files and download takes much longer

Is this happening every time, or just in some cases??

Have you tried same magnet/file again, that triggered that error???

**Comment by skbaji6** [comment]
> Can I know the value of your `EDIT_SLEEP_TIME_OUT` ?🤔
> 
> Edit-- I read that😐
> That error if from telegram...may be a lot of process is running, may be that causing the edit error....😓

Yeah that might be the cause. but you need to optimize the code to handle the RPC error and need to introduce the retry logic you just cant ignore the RPC error and cant depend upon the sleep time every time.

**Comment by gautamajay52** [comment]
> > Can I know the value of your `EDIT_SLEEP_TIME_OUT` ?🤔
> > Edit-- I read that😐
> > That error if from telegram...may be a lot of process is running, may be that causing the edit error....😓
> 
> Yeah that might be the cause. but you need to optimize the code to handle the RPC error and need to introduce the retry logic you just cant ignore the RPC error and cant depend upon the sleep time every time.

Those all codes are from base repo.😬
And I have never touched those.😓

**Comment by sevvian** [comment]
> > > Can I know the value of your `EDIT_SLEEP_TIME_OUT` ?thinking
> > > Edit-- I read thatneutral_face
> > > That error if from telegram...may be a lot of process is running, may be that causing the edit error....sweat
> > 
> > 
> > Yeah that might be the cause. but you need to optimize the code to handle the RPC error and need to introduce the retry logic you just cant ignore the RPC error and cant depend upon the sleep time every time.
> 
> Those all codes are from base repo.
> And I have never touched those.

I think to further explain the issue , please find the screenshot below, you could see that aria2 continued downloading and finally finished, but (from the log) Rclone started and failed failed copying because Rclone got triggered while aria was still downloading .
[Screenshot from bot](https://pasteboard.co/JmELFqu.png) (after few hours after rclone failed)

If you can implement the fix where rclone should  only be triggered only if Aria completes (ignoring any pyrogram based error messages like 400 bad message) and says job is completed through some rpc call using aria RPC, then this problem can be avoided i guess

**Comment by gautamajay52** [comment]
> > > > Can I know the value of your `EDIT_SLEEP_TIME_OUT` ?thinking
> > > > Edit-- I read thatneutral_face
> > > > That error if from telegram...may be a lot of process is running, may be that causing the edit error....sweat
> > > 
> > > 
> > > Yeah that might be the cause. but you need to optimize the code to handle the RPC error and need to introduce the retry logic you just cant ignore the RPC error and cant depend upon the sleep time every time.
> > 
> > 
> > Those all codes are from base repo.
> > And I have never touched those.
> 
> I think to further explain the issue , please find the screenshot below, you could see that aria2 continued downloading and finally finished, but (from the log) Rclone started and failed failed copying because Rclone got triggered while aria was still downloading .
> [Screenshot from bot](https://pasteboard.co/JmELFqu.png) (after few hours after rclone failed)
> 
> If you can implement the fix where rclone should only be triggered only if Aria completes (ignoring any pyrogram based error messages like 400 bad message) and says job is completed through some rpc call using aria RPC, then this problem can be avoided i guess

 Yeah,  I noticed that from rclone error message in logs.😐
